### PR TITLE
Remove circular dependencies in VelocityServer where sensible

### PR DESCRIPTION
**Pull request by A248**

### CODE REVIEW

File: proxy/src/main/java/com/velocitypowered/proxy/Velocity.java [hunk 1]
```diff
@@ -54,14 +60,23 @@ public static void main(String... args) {
 
     long startTime = System.currentTimeMillis();
 
-    VelocityServer server = new VelocityServer(options);
+    VelocityPluginManager pluginManager;
+    VelocityEventManager eventManager;
+    VelocityConsoleCommandSource console;
+    VelocityServer server = new VelocityServer(
+            (pluginManager = new VelocityPluginManager()),
```
Comment (kashike) [inline proxy/src/main/java/com/velocitypowered/proxy/Velocity.java]: Can you move these assignments to the variables directly, and reference them normally here instead?

**Review by kashike — CHANGES_REQUESTED** [review]

File: proxy/src/main/java/com/velocitypowered/proxy/Velocity.java (hunk 1 above)
Comment (A248) [reply proxy/src/main/java/com/velocitypowered/proxy/Velocity.java]: It will be less beautiful then. I will do it, though

**Comment by A248** [comment]
I'm not sure why the tests are failing. They succeed locally. The failure is related to the CommandManager tests.

**Comment by astei** [comment]
I think @hugmanrique needs to look into that.

**Comment by hugmanrique** [comment]
The failing test is incorrect and the actual expected behavior was introduced by the recent Brigadier merges. It has to do with considering both the hint suggestions and the regular suggestions when a command has both. I'm not at the computer right now, but the test can be `@Disabled` for now.

Sorry for the breakage 😓

**Pull request closed without merging**